### PR TITLE
Logstash simple installing

### PR DESCRIPTION
**Pull request by Karina5005**
#1

### CODE REVIEW

**Comment by Karina5005** [comment]
Pull created by EnergyNick at 2022-11-11T13:26:54Z

File: tasks/configuration.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
```
Comment (a-alex-l) [inline tasks/configuration.yml:1]: It's better to move this task to handlers. Isn't it?

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (a-alex-l) [inline defaults/main.yml:1]: Where does certificate configuration present on this pr?

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [inline tasks/configuration.yml:1]: I can't see any files above in this pr

File: defaults/main.yml (hunk 2 above)
Comment (a-alex-l) [inline defaults/main.yml:1]: Where is a beats configuration?

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [inline tasks/configuration.yml:1]: There aren't any files

Comment (a-alex-l) [inline tasks/configuration.yml:1]: Do we really need to notify handlers if we have `--config.reload.automatic` option in service unit? I don't know, need testing

Comment (a-alex-l) [inline tasks/configuration.yml:1]: Where is this template?

File: templates/logstash.service.j2 [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# {{ ansible_managed }}
```
Comment (a-alex-l) [inline templates/logstash.service.j2:1]: Add `# {{ ansible_managed  }}` string to any template managed by Ansible. Read more at [config docs](https://jpmens.net/2020/09/29/using-ansible-managed/).

Note: don't add any dynamic value into that variable

File: tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (a-alex-l) [inline tasks/install.yml:1]: Why we add user to that group?

File: templates/logstash.service.j2 (hunk 3 above)
Comment (a-alex-l) [inline templates/logstash.service.j2:1]: It's really bad practice to hardcode variables. Add them to defaults and put here

File: defaults/main.yml (hunk 2 above)
Comment (a-alex-l) [reply defaults/main.yml:1]: This is template for next update, used from other ansible role

File: templates/logstash.service.j2 (hunk 3 above)
Comment (a-alex-l) [reply templates/logstash.service.j2:1]: This group should allow access to logstash to use system monitor files, I believe this is necessary for some internal reasons.

File: tasks/install.yml (hunk 4 above)
Comment (a-alex-l) [reply tasks/install.yml:1]: I don't have access to that file(

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [reply tasks/configuration.yml:1]: Oh, i don't push it, strange 🤔

Comment (a-alex-l) [reply tasks/configuration.yml:1]: fixed

Comment (a-alex-l) [reply tasks/configuration.yml:1]: fixed

File: defaults/main.yml (hunk 2 above)
Comment (a-alex-l) [reply defaults/main.yml:1]: fixed

File: templates/logstash.service.j2 (hunk 3 above)
Comment (a-alex-l) [reply templates/logstash.service.j2:1]: My mistake, delete this unused file

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [reply tasks/configuration.yml:1]: Moved to handler

Comment (a-alex-l) [reply tasks/configuration.yml:1]: Fixed

Comment (a-alex-l) [reply tasks/configuration.yml:1]: Ok, test this and rewrite to one specific style

File: defaults/main.yml (hunk 2 above)
Comment (a-alex-l) [reply defaults/main.yml:1]: Let's remove it from this pr

File: tasks/install.yml (hunk 4 above)
Comment (a-alex-l) [reply tasks/install.yml:1]: Wrong url, fixed

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [inline tasks/configuration.yml:1]: You may replace this two handler with single one as I mentioned above

File: tasks/create_users.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (a-alex-l) [inline tasks/create_users.yml:1]: The same as above

Comment (a-alex-l) [inline tasks/create_users.yml:1]: Are you really going to create user role on elasticsearch using logstash? I suppose it's better to delegate role preparation to elasticsearch ansible role

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (a-alex-l) [inline tasks/main.yml:1]: Use `systemd` instead

File: molecule/default/converge.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (a-alex-l) [inline molecule/default/converge.yml:1]: Why is it here? Could you add this variable in vars below?

File: handlers/main.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (a-alex-l) [inline handlers/main.yml:1]: You may reduce this handler by switching `restart logstash necessarily` handler from `service` to `systemd`.
We actually don't have to support old OS versions that are supplied without systemd.

Comment (a-alex-l) [inline handlers/main.yml:1]: We should avoid `when` condition inside handler. It's better to handle condition in tasks

File: tasks/plugins.yml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+---
```
Comment (a-alex-l) [inline tasks/plugins.yml:1]: rewrite (here and above) using modern ansible style

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [inline tasks/configuration.yml:1]: Future request: think please about how to dynamically add any filters that you need. At the current state You can't rewrite existent filters as well as add new

File: handlers/main.yml (hunk 8 above)
Comment (a-alex-l) [reply handlers/main.yml:1]: Ok, change behaivour

File: molecule/default/converge.yml (hunk 7 above)
Comment (a-alex-l) [reply molecule/default/converge.yml:1]: Fixed

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [reply tasks/configuration.yml:1]: Reworked

File: tasks/main.yml (hunk 6 above)
Comment (a-alex-l) [reply tasks/main.yml:1]: Fixed

File: tasks/create_users.yml (hunk 5 above)
Comment (a-alex-l) [reply tasks/create_users.yml:1]: Changed behaivour to "Additional functionality" of the role

Comment (a-alex-l) [reply tasks/create_users.yml:1]: Fixed same as above

File: handlers/main.yml (hunk 8 above)
Comment (a-alex-l) [reply handlers/main.yml:1]: Ok, rework handlers

File: tasks/plugins.yml (hunk 9 above)
Comment (a-alex-l) [reply tasks/plugins.yml:1]: Didn't understand you a bit. Can you please decipher the phrase, what is "modern ansible style"?

File: tasks/configuration.yml (hunk 1 above)
Comment (a-alex-l) [reply tasks/configuration.yml:1]: For now - nothing, therefore delete it.
Add later, if nessesary

File: tasks/plugins.yml (hunk 9 above)
Comment (a-alex-l) [reply tasks/plugins.yml:1]: Use extended yaml view. Compare the following

`chdir={{ logstash_dir }}`

```
  args:
    chdir: {{ logstash_dir }}
```

File: handlers/main.yml (hunk 8 above)
Comment (a-alex-l) [inline handlers/main.yml:1]: `daemon-reload`?

Comment (a-alex-l) [reply handlers/main.yml:1]: Fixed

**Comment by Karina5005** [comment]
Assignee is {'login': 'EnergyNick'}

**Pull request closed without merging**